### ISSUE TITLE
Add wildcards or custom filters to the hidden dot files option.

### ISSUE THREAD

**Issue by KosmosisDire** [files-community/Files#8423]
### What's the Problem?

In Linux dot files are usually hidden, but there are also other cases where usually hidden files are not hidden in Windows. My specific problem is that I have a Linux network drive mounted on my computer. I store lots of word files on this drive and word files create a temporary "~$filename.docx" file for every word file. These are usually hidden on windows, but because it's mounted from Linux using SSHFS nothing is hidden.

### Solution/Idea

Add more advanced options under the "hide dot files" option to allow custom wildcards or filters. I know there was talk about this under [this](https://github.com/files-community/Files/issues/2587) issue but I think it needs a separate issue for proper discussion.

### Alternatives

Keep it sweet and simple with just the dot option.

### Priorities

| Capability | Priority |
| :---------- | :------- |
| This proposal will allow users to hide files that **begin** with a specific string | Must |
| This proposal will allow users to hide files that **end** with a specific string | Must |
| This proposal will allow users to hide files that **contain** a specific string | Should |
|This proposal will allow users to specify regex to hide files | Could |

### Files Version

N/A

### Windows Version

N/A

### Comments

Here is an example of some pesky files that should be hidden but aren't:
![Temporary files that should be hidden](https://user-images.githubusercontent.com/39423700/154810161-9610eea6-8597-4686-a969-55e266c38acd.png)

Here is a quick mockup of how the option could be changed to allow this. It Is really inconsitent though since every other option here is just an on off toggle:
![Hidden Files Mockup 2](https://user-images.githubusercontent.com/39423700/154811932-d459b867-16e6-4140-b7c4-7e029ee0aec3.png)

And with it open:
![Hidden Files Mockup](https://user-images.githubusercontent.com/39423700/154811948-76f96bf4-779b-4b15-9439-438190eebe93.png)

**Comment by KosmosisDire** [comment]
Here is a better mockup:
![Hidden Files Mockup 3](https://user-images.githubusercontent.com/39423700/154814237-9cbd3e29-f49c-4066-b3a1-be897f32eb5a.png)

**Comment by Josh-65** [comment]
Insided of custom filter since i dont think there many files to hide adding ~$ to the list should be enough

**Comment by KosmosisDire** [comment]
@Josh-65 Yeah, that's true, although it would be just "~". The second file in my image doesn't have a "$". 

Someone in the linked issue also said they worked with files that were "__file" and "_file" that should be hidden:
>I work a lot with files which start with _ or __ these files clutter some folders a bit, so for me it would certainly be useful. With Regex I could even hide files with specific endings. The usecases are endless.

So it could filter these symbols:
**~**
**_**
 **.**

However, they would have to be togglable individually at the very least. As for hiding things like the lost and found folder, or normally-hidden system files, that wouldn't be possible. It would fix most of the most common use cases though

**Comment by Aziks0** [comment]
I have the same request but for the `desktop.ini` files that are created in folders with custom icons. They drive me crazy.

**Comment by Josh-65** [comment]
Asking @yaichenbaum for approval.

**Comment by yaichenbaum** [comment]
Wouldn't `.docx` hide all Word documents?

**Comment by KosmosisDire** [comment]
@yaichenbaum If you are referring to the mockup I did, that was purely an example, I don't think you'd want to do that. But there are other more useful endings that could be put there. Like ends with .ini, or starts with $~, etc...
